### PR TITLE
Fix degree type free text bug

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

If you select a degree level like Bachelors, then enter a subject, but then enter a different type of bachelor as free text (eg "Bachelor of Things") then the summary table doesn't display both questions as separate rows, just "Degree type: Bachelor of Things".

Instead it should behave as if you selected Bachelor of Arts or another type, and summarise it as:

Degree type: Bachelor
Type of Bachelor degree: Bachelor of Things

## Data migrations

I decided to run manually the migration as will backfill approx 50k records in the database. Added the dry run to get more context before running it.

## Guidance to review

1. Does it work properly for newly added degrees?

## Link to Trello card

https://trello.com/c/dt10YnYp/55-degree-type-free-text-bug

### CODE REVIEW

**Comment by tomas-stefano** [comment]
@frankieroberto is all good now! The bug was happening before this PR but I was happy to have to fix them all! ✌️ 🚀

**Comment by frankieroberto** [comment]
@tomas-stefano thanks, didn't realise that was an existing bug! 🤦‍♂️ 

I just checked the review app again, and the first wording bug is fixed, but the second bug still seems to happen...? (ie typing "Diploma of Life" still gets converted into "Level 6 Diploma" when returning to the page from the review)

**Comment by tomas-stefano** [comment]
@frankieroberto I couldn't reproduce the bug on the review app. Can we talk in Slack so we can talk about this bug, please?

**Review by frankieroberto — APPROVED** [review]
Re-tested in the review app, and both bugs are fixed. Thanks!

File: app/services/data_migrations/backfill_qualification_level.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+module DataMigrations
+  class BackfillQualificationLevel
```
Comment (cwrw) [inline app/services/data_migrations/backfill_qualification_level.rb:2]: Could we separate the backfill into another pr so it can be merged and we can run separately?

Comment (tomas-stefano) [reply app/services/data_migrations/backfill_qualification_level.rb]: Sure, I removed the backfill from this PR. I'll open another one.

File: app/components/candidate_interface/degree_new_review_component.rb [hunk 2]
```diff
@@ -271,15 +271,19 @@ def generate_action(degree:, attribute: '')
...
       return if degree.qualification_type.nil?
 
-      reference_data = DfE::ReferenceData::Degrees::TYPES.some_by_field(:name).keys.select { |type| degree.qualification_type.include?(type) }
-      if reference_data.present?
-        degree.qualification_type.split.first
+      if degree.qualification_level.present?
+        DegreeWizard::QUALIFICATION_LEVEL[degree.qualification_level]
+      else
+        reference_data = DfE::ReferenceData::Degrees::TYPES.some_by_field(:name).keys.select { |type| degree.qualification_type.include?(type) }
+        degree.qualification_type.split.first if reference_data.present?
```
Comment (JonF1) [inline app/components/candidate_interface/degree_new_review_component.rb:278]: if we're retaining this then we may have a mismatch. E.g this will render Bachelor for type of degree row instead of Bachelor degree. The key in the row will be Type of Bachelor degree, instead of Type of bachelor degree. To replicate, type in Bachelor of Arts in the another qualification text field on degree level page. Small nitpick tho.

**Review by JonF1 — APPROVED** [review]

File: app/components/candidate_interface/degree_new_review_component.rb (hunk 2 above)
Comment (tomas-stefano) [reply app/components/candidate_interface/degree_new_review_component.rb:278]: I see. Is this is a blocker to merge?

**Review by cwrw — COMMENTED** [review]
looks good, I noticed however that when attempting to change `Type of bachelor degree` when it's manually entered the radio box is not selected on the page:
Screenshot 2022-05-30 at 20 28 25

**Comment by tomas-stefano** [comment]
@cwrw that's interesting, I can't reproduce the bug 🤔 

Screenshot 2022-05-31 at 01 07 31

This when I reload the page everything is selected 🤔 

Can you show me at some point today, please?

File: app/components/candidate_interface/degree_new_review_component.rb (hunk 2 above)
Comment (tomas-stefano) [reply app/components/candidate_interface/degree_new_review_component.rb:278]: Fixed!

**Review by cwrw — APPROVED** [review]
